### ISSUE TITLE
The official network wishlist for Riot

### ISSUE THREAD

**Issue by ara4n** [matrix-org/matrix-appservice-irc#208]
Right now we're not letting people bridge to arbitrary networks of their choice, but curating the bridges we set up (in order to ensure that we connect at scale to those networks in a way that doesn't upset them, and get permission first, setup I:lines etc, add the networks to Riot's official config.json etc).

So let's have an official backlog here for the networks that folks want to be connected to.  Ones i've already heard about are:

 * Freenode
 * Moznet
 * W3C
 * OFTC
 * Snoonet  (alt Reddit)
 * Foonet  (XKCDland)
 * geekshed  (Jupiter Broadcasting)
 * Rizon
 * IRCnet 

Folks: if you have particular networks you want bridged, please add them here.

**Comment by Cadair** [comment]
@drewleonard42 is the one you wanted on this list?

**Comment by drewleonard42** [comment]
No, I think it was synirc.

**Comment by eternaleye** [comment]
I'd like Lunarnet, AKA Svartalfheim - there's a pair of rooms I was in pretty much constantly, but since the bridge running on `yochat.biz` went down a while ago I've been offline there :(

**Comment by TheCapsLock** [comment]
It would be sweet to have Geeknode (http://www.geeknode.org/) in supported servers list for IRC integration

**Comment by Mikaela** [comment]
I would like to see a bridge for [PirateIRC](http://pirateirc.net/).

> Please note that despite the name this is an IRC network for political
> parties and this IRC network does NOT advocate "pirating" software. This
> network is for legal discussions and chat only. If you wish to contact the
> network administrators to report any illegal activity please email:
> admins@pirateirc.net

I heard that Pirate Party Australia (they are PPAU, aren't they) are looking into it, but I never heard anything further and have no idea whom to contact (even if I haven't investigated their channels and asked around).

**Comment by Ralith** [comment]
I'd like to see synirc and EsperNet.

**Comment by sbts** [comment]
It would be good to add irc.perl.org to the list

**Comment by neurocis** [comment]
FYI I have setup a GeekShed bridge for the Jupiter Broadcasting channel. If there is enough interest I would be willing to expand it to all geekshed channels and move it to a more permanent home:

#jupiterbroadcasting:qubed.net

Also thanks to @lukebarnard1 for his help during setup.

**Comment by aakord** [comment]
Well, [EFnet ](https://en.wikipedia.org/wiki/EFnet) was there in internet back to '90 (about ~8 before google); so I guess they won't get easily upset if you add them :+1:

**Comment by pztrn** [comment]
Quakenet, please? :)

**Comment by vext01** [comment]
I'd really appreciate a link to the blitzed IRC network. Thanks.

**Comment by locutis-of-borg-1999** [comment]
Here is a comprehensive list, alphabetical:  http://irc.netsplit.de/networks/  

That page contains a link within the same site, which ranks it's "Top 100"  IRC networks that are "competitive".  Freenode, for example, is NOT ranked there.  So, the Alphabetical list is a better reference.   

Anyway, the "top" 10, not already listed here, that have MANY users: 
Undernet, Rizon, ChLame, IRC-Hispano, DALnet, OltreIrc, ExplosionIRC, 	Abjects, P2P-NET, GameSurge

The more people connected via Matrix, the better.  One "selling point" is that if they come over to the Matrix side, they do NOT need a "bouncer" (BNC/ZNC)!!!  


I suggest the O.P. post in alphanumeric order, first already bridged, then the hopefuls, which should be easier for all, as the list grows.  

Also, maybe some IRCOps/SysAdmins might like control of their own Matrix server instance, or not.

Thanks.

**Comment by locutis-of-borg-1999** [comment]
I would like to see Undernet & DALnet

Thanks.

**Comment by jjardon** [comment]
GIMPNet (irc.gnome.org) please?

**Comment by albfan** [comment]
Missing GIMPNet too.

**Comment by fuzzy76** [comment]
Those already implemented should really be in the repo wiki or the readme. I have been searching for a list of them for a long time. :-/

**Comment by Mikaela** [comment]
> Those already implemented should really be in the repo wiki

It's a wiki so feel free to edit it :)

It would be nice if in addition to the name and prefix the appservice name was listed so the end-user FAQ could refer there.

**Comment by albfan** [comment]
What would be the roadmap to integrate gimpnet (irc.gnome.org) can be help in any way?

**Comment by elnygren** [comment]
According to wikipedia, IRCnet is the #2 largest network. Missing it is currently a bit of a dealbreaker for me (and probably other people as well). 

What could be done to get it ASAP? Could I help in some way?

https://en.wikipedia.org/wiki/Internet_Relay_Chat#Networks

**Comment by fuloating** [comment]
I have been asking about IRCnet aswell. I'm not that apt with the technicalities, but I'll share this slightly edited log from some months back when I was investigating. Nothing special there but could be something of relevance. (I shared it back then too somewhere after asking if it was ok) It's from IRCnet #ircnet. The eu.irc6.net operator uepsie can be contacted on ircnet, and irc.atw-inter.net operator BR likewise. They both also idle on IRCnet #ircnet.

**Comment by Powersource** [comment]
* irc.overthewire.org
* irc.afternet.org
* irc.tribalwar.com

Btw thanks for all of this, I love the IRC communities but the clients have always been horrible (at least from my perspective).

**Comment by Kegsay** [comment]
Filed a status update on the OP.

**Comment by vext01** [comment]
For the record, Blitzed IRC have said they are happy to have their network bridged to matrix. The person to speak to is `dg` in #blitzed on blitzed IRC. He requests that we let them know when a bridge has connected so that they can identify it (and presumably review the connection limits).

**Comment by ericblank** [comment]
I would like to see Indymedia IRC and 2600 IRC

irc.indymedia.net
irc.2600.net

**Comment by Croydon** [comment]
https://gamesurge.net/?language=English would be nice

**Comment by lub** [comment]
https://www.euirc.net/en/
https://help.twitch.tv/customer/portal/articles/1302780-twitch-irc

**Comment by locutis-of-borg-1999** [comment]
According to their websites, as of 2013, both DALnet and Undernet offer ipv6 connectivity.  

https://www.dal.net/news/shownews.php?id=122  

http://help.undernet.org/topics.php?what=newin12#05

**Comment by fladd** [comment]
https://www.mixxnet.net/ would be nice to have.

**Comment by MilkManzJourDaddy** [comment]
I think Rizon had some better features, maybe DALnet, too.  I might have some old contacts on those, and Undernet.  And they seem to have many users.  So:  
- Undernet
- DALnet
- Rizon

**Comment by heftig** [comment]
I'd like to see [IRCHighWay](https://irchighway.net/) bridged, too.

**Comment by BytesAndCoffee** [comment]
Please remove your bot from channels on Foonetic, specifically #xkcd- channels, unauthorized bots are prohibited

**Comment by ptman** [comment]
@BytesAndCoffee (note, I'm not talking on behalf of matrix.org)

Would it be possible to authorize the bot? The way matrix-appservice-irc works is, that the bot can be on channels and receive messages so that it can deliver them to matrix users that aren't active on the IRC channel. This saves connections from the bridge to the IRC server. And also allows users to catch up on messages while they are away.

If not, would it be ok if the bot doesn't join channels? AFAIK the bot cannot be disabled completely, but it can be set to not join channels so that every user joins via a separate IRC connection.

**Comment by BytesAndCoffee** [comment]
It may be possible to ask the ops to authorize the bot, but you need to fix a bug first. #xkcd uses an open topic, where users can prepend freely. Your bot (enick_204) replicates the topic change unnecessarily, after fixing that, you may be able to push for official bot status, but given the channel governance model, that is unlikely. 

NOTE: I am not a channel moderator and purely speak from experience

**Comment by ara4n** [comment]
@BytesAndCoffee just to clarify: there was a misconfig of some kind which caused this enick_204 to join. there was no intention whatsoever to join bots to #xkcd; i was just using Matrix as a normal IRC bouncer to connect.  I'm investigating now.

I'm assuming there aren't problems with using IRCCloud or ZNC-style bouncers to connect (and this is the only intended mode of operation for matrix in this instance).

**Comment by Mikaela** [comment]
PirateIRC:

> it's ok to add new gateways as long as a) they don't affect security and b) they can be banned if there are problems like mass spamming channels etc..

> Can you send me the IP of gateway, I'll add it some rights to have enough connections thru single IP address.

**Comment by Mikaela** [comment]
[Furnet](http://www.furnet.org/wiki/index.php/Main_Page) / irc.furnet.org (or ipv6.furnet.org) would also be nice to have bridged to Matrix.

**Comment by Mikaela** [comment]
PirateIRC has been bridged a few days ago and the details can be found from [the wiki](https://github.com/matrix-org/matrix-appservice-irc/wiki/Bridged-IRC-networks#hosted-by-third-party) (`#pirateirc_#channel:diasp.in`).

*I have been trying to request the issue to be edited at Matrix, but as no one ever did it, I think it's best that I add this comment.*

**Comment by grahamperrin** [comment]
> EFnet : irc.efnet.org 

:+1: 

Today in irc://irc.efnet.org/#freebsd-xorg I asked: 

>> Random question (if it has been asked before, sorry): ever thought of moving this channel from EFnet to freenode?

The reply: 

> nope, most of the bsd channels (#bsdports, #bsddev, #bsdcode, #bsdmips, etc) are on EFnet. #kde-freebsd is the odd one out on freenode

Not just one, other BSD-related channels on freenode include: 

- ##bsd
- #freebsd
- #freebsd-bugs
- #freebsd-commits
- #freebsd-labs
- #freebsd-ports
- #freebsd-social
- #freebsd-src
- #freebsd-vbox
- #freebsd-virtualbox
- #freenas
- #ghostbsd
- #knoxbug
- #openbsd
- #unitedbsd

Still, there's my vote for a bridge to EFnet becuase I doubt that there'll be a shift away (to freenode) for the five channels of which I'm aware: 

- #bsdcode 23 ops, 94 total
- #bsddev 23 ops, 95 total
- #bsdmips 25 ops, 99 total
- #bsdports 41 ops, 162 total
- #freebsd-xorg 9 ops, 55 total

**Comment by Kegsay** [comment]
> I have been trying to request the issue to be edited at Matrix, but as no one ever did it, I think it's best that I add this comment.

Done. Thanks @Mikaela

**Comment by turt2live** [comment]
Bridging irc.spi.gt would be nice. Their terms of use do not support the bridge, however they may be willing to negotiate a bridge. They also have IPv6 support :)

**Comment by Xe** [comment]
Please set up irc.ponychat.net.

**Comment by vext01** [comment]
I tried to bridge blitzed under my own steam tonight, but I was scared off by lots of scary C++ errors when running npm. I think I'd prefer the experts to do the bridging, since it seems to be quite involved and sensitive to software versions :(

**Comment by static-void** [comment]
Registering interest in bridging blitzed here

**Comment by jflory7** [comment]
I would like to see the [SpigotMC IRC network](https://irc.spi.gt/) added. All servers use IPv4 and IPv6, and you can see the full server list [here](https://irc.spi.gt/servers.php). Generally, @md-5 is the point of contact there, but you can drop an email to staff [at] spigotmc [dot] org too.

**Comment by wakest** [comment]
just commenting to bump the bridging of EFnet. Been really wanting to follow the ArchiveTeam's room from riot.

**Comment by 14mRh4X0r** [comment]
IRCNet has several open servers listening on IPv6, as can be seen by resolving servers on [this list](http://irc.tu-ilmenau.de/all_servers/?focus=1&open=1).
That said, we at [SNT](http://www.snt.utwente.nl/) (the organisation behind openirc.snt.utwente.nl) are going to host our own Matrix bridge, but probably only once Dendrite is production-ready.

**Comment by t3chguy** [comment]
QuakeNet appears to have some ipv6 support, may be good enough? (not my need)

**Comment by elouin** [comment]
To have Quakenet would be awesome, since its one of the biggest IRC Networks(If not the biggest)...

**Comment by Kegsay** [comment]
> QuakeNet appears to have some ipv6 support

```
$ host irc.quakenet.org
irc.quakenet.org has address 66.225.225.66
irc.quakenet.org has address 83.140.172.211
irc.quakenet.org has address 83.140.172.212
irc.quakenet.org has address 80.65.57.26
irc.quakenet.org has address 128.39.65.226
irc.quakenet.org has address 83.140.172.210
irc.quakenet.org has address 91.217.189.21
```

Though it looks like there is a solitary IPv6 server on:
```
$ host irc.ipv6.quakenet.org
irc.ipv6.quakenet.org has IPv6 address 2001:16d8:aaaa:2::1337
```

Will update accordingly.

**Comment by Kegsay** [comment]
Updated OP. General status update:

I'm generally happy with the state the IRC bridge is in currently. It's not perfect by any means (as the 100 open issues will attest), but I'm not aware of any critical bridge-destroying bugs out there anymore. Previously, we encountered problems for very large networks (ala Freenode) whereby performance became a real problem. After a lot of profiling and bug fixes, this doesn't seem to be the case anymore.

That being said, we are encountering occasional performance issues on the main matrix.org homeserver which *is* affecting our ability to deploy new IRC networks. Coupled with that, we're quickly approaching the point where we bridge so many IRC networks that our existing ops setup isn't working that great for us (deploying new versions, allocating ports / IPv6 blocks, metrics, logs etc become a pain when you have to do it N times). We're going to need to think about how to resolve these issues so we can scale out networks in a sustainable way.

It's not all bad news though. We recently enabled Rizon and EsperNet.

**Comment by locutis-of-borg-1999** [comment]
i wanted rizon  so thanks

**Comment by locutis-of-borg-1999** [comment]
butt also dalnet and undernet are kinda large with ipv6 like i showed so can they at least be on the wishlist with the lightening bolt

**Comment by locutis-of-borg-1999** [comment]
https://www.dal.net/news/shownews.php?id=122

http://help.undernet.org/topics.php?what=newin12#05

**Comment by locutis-of-borg-1999** [comment]
http://irc.netsplit.de/networks/top100.php

**Comment by locutis-of-borg-1999** [comment]
that links shows that there might be 500 ponychat users but here in #208 they seem to matter more than the 20000 concurrent on undernet or 7000 concurrent on dalnet

**Comment by ara4n** [comment]
a) ponychat is bridged by the community, not by us. b) we have limited capacity on the matrix.org HS, which is precisely why we haven't bridged many larger networks like undernet yet.

**Comment by locutis-of-borg-1999** [comment]
oh thanks for splaining i thought it was the power of adult male brony/bronie fans of the my little pony show showing more influence but maybe if as the above links show that undernet and dalnet have ipv6 we could give a go of adfing them to the toppost/op until matrix can handle the extra load thanks

**Comment by Xe** [comment]
@ara4n I pulled down the ponychat bridge, it was difficult to debug or introspect given the service is written in node.js.

**Comment by ara4n** [comment]
okay. since those days the bridge has got way better fwiw, and even has a manhole for realtime introspection and debugging (but in general one should never need to do so, unless dealing with something huge like FN)

**Comment by jaller94** [comment]
Paranode (irc6.paranode.net) would be awesome as it hosts the local Freifunk community of Cologne, Bonn and the surrounding area.

**Comment by silkeh** [comment]
There is now a bridge to IRCnet: `#_ircnet_#channame:irc.snt.utwente.nl`

**Comment by staticsafe** [comment]
I'd like to request a bridge to irc.entropynet.net. All servers are IPv6 enabled and that RR has AAAA records.

Let me know what exempts I have to add.

**Comment by daurnimator** [comment]
Could you please add irc.hashbang.sh?

**Comment by feld** [comment]
irc.linpro.no would be nice. That's where Varnish is developed. 👍

**Comment by henk717** [comment]
Here is another vouch for GameSurge.net, these host the channels of the HL2 server community.
Main interest is for #sourcemod and #nfo-support.

**Comment by zeenix** [comment]
I recently switched to using matrix for my IRC needs. I can connect to GNOME, Freenode and Mozilla IRC so most of my needs are already satisfied. However, there is one channel on IRCNet that I need to connect to and I wouldn't want to run two different chat clients just for being able to connect to one additional channel. So here is a big :+1: for IRCNet.

**Comment by Kegsay** [comment]
@zeenix - https://github.com/matrix-org/matrix-appservice-irc/issues/208#issuecomment-311629569

**Comment by zeenix** [comment]
@Kegsay hm.. weird. I tried #_ircnet_#meritahti:irc.snt.utwente.nl and #_ircnet_#meritahti:irc.ircnet.com and ended up on empty channels, although I can see from xchat that they are not empty.

**Comment by silkeh** [comment]
@zeenix you can join `#ircnet:matrix.org` for support.

**Comment by rodneyrod** [comment]
Has any progress been made on bridging Furnet yet?

**Comment by locutis-of-borg-1999** [comment]
hey howzabout we get nets with moor peeps b4 bowing 2 furry n mlp brony/bronie freaks eh? thur are lotsa sex chans on matrix now.  if thes small groops want to mix thur matrix n irc they prob got control on irc net.  let em run a matrix and irc bridge.  the pub matrix is for big irc that needs big machines nd maybe ask irc powers.  butt mabe ip6 dont kneed iline if all get each owned ip addy. butt sum big nets r lef out

**Comment by eggy** [comment]
Please feel free to reach out to me for adding an officially supported bridge to IRCHighWway (irc.irchighway.net). I'm part of the appropriate admin team to help with implementation (on a side note, I started looking into setting up a bridge to IRCHighWay through my own hosted HS).

**Comment by Spydar007** [comment]
@rodneyrod I currently run a private bridge for Furnet. I have requested I:lines but the opers who have the ability to set I:lines do not respond to the other opers and therefore I am getting nowhere, so I doubt anyone else will either. You can message me on Matrix at @Spydar007:spydar007.com if you wish to use my private bridge however, as I currently have some spare connections available.

**Comment by locutis-of-borg-1999** [comment]
K, I am not all the time the best with writings but I try.  Anyone else ?  No, huh?  @eggy maybe get support on ` #irc:matrix.org ` which also goes by bridge 2 Freenode ` #irc-matrix ` yeah?

**Comment by locutis-of-borg-1999** [comment]
Now. Can some tell on some IRC nets with ipv6 if iline is a thing or not ?  If a net has ipv6 and do not care to need iline as well it should be easy but I ask not tell

**Comment by Spydar007** [comment]
@locutis-of-borg-1999 Yes, an I:line is an exemption block for both IPv4 and IPv6. Generally, networks will define a set number of users allowed per subnet. For example, they might say that you are allowed 20 users per /48. It is a good idea to request an I:line for the bridge regardless of IPv4 or IPv6. You should also always seek permission from the operators of the network before running a bridge regardless of any other criteria you may or may not fulfil.

**Comment by andrewshadura** [comment]
Can we please have bynets.org bridged? There's a UTF-8 port, the default is a non-UTF-8 encoding; the network supports non-Latin characters in the nicknames, but they should be from the CP1251 range.

**Comment by mbe24** [comment]
It was already mentioned in late 2016 and mid 2017, but please: Quakenet.

**Comment by Serkan-devel** [comment]
Can Ponychat IRC be added?

**Comment by DinizCabreira** [comment]
IRC-Hispano is a huge spanish-language (mostly) network. Having it on board might drag a lot of users into Matrix and Riot.

According to their statistics, they have between 15k and 29k simultaneous users, and nearly 3 million users per month.

**Comment by rogersachan** [comment]
IRCnet should be crossed off the list. They already have a bridge hosted by the network, as seen on the wiki.

**Comment by dalecarlian** [comment]
+1 Quakenet :)

**Comment by grahamperrin** [comment]
Please vote on existing comments (don't add +1 comments).

**Comment by grahamperrin** [comment]
`irc.w3.org` maybe? 

Or is the w3 case under the umbrella of the W3**C** comment? 

> … removed W3C support due to them dropping their IPv6-enabled IRCd. …

https://www.w3.org/Project/IRC/

**Comment by Matrixcoffee** [comment]
In case anyone's wondering what's changing in the top post, https://github.com/Matrixcoffee/track-issue-208 now exists.

(Not much to see yet, but if you come upon this later there should be more history.)

**Comment by c-e-p-x-u-o** [comment]
I vote for irc.perl.org!

**Comment by all-the-good-ones-are-gone** [comment]
I'll throw one in for sorcery.net.

**Comment by sdier** [comment]
slashnet

**Comment by sjn** [comment]
Yeah, irc.perl.org access would be lovely.

What needs to be in place (other than IPv6 access) for this to happen?

**Comment by raboof** [comment]
I'd love to see irc.smurfnet.ch bridged. They seem to have an IPv6 endpoint available.

**Comment by DanitiKain** [comment]
spotchat.org many of the linux distros are starting to use this irc network one of them being Linux Mint distro.

**Comment by lord-re** [comment]
What's required on the ircd side to bridge with matrix ? Just upping the userlimits on a specified IP ?
As a netadmin of geeknode.org i would gladly do this to bridge with matrix.

**Comment by Matrixcoffee** [comment]
@lord-re At the basic level, yes. Are you also offering to run the bridge?

**Comment by lord-re** [comment]
@Matrixcoffee if it runs easily on a linux host, why not ?

**Comment by Matrixcoffee** [comment]
@lord-re Since you also need to run your own Matrix server with the bridge, you'll have a non-trivial amount of extra maintenance load on your hands. It _is_ the only self-service way, though. Maybe someone else you trust wants to run it for you.

I don't think the Matrix team is "hiring" new networks at the time, as the bridges and server are overloaded enough. However, if you want to formally request bridging as network staff, your best bet is to get the contact address off of matrix.org, and email them.

**Comment by lord-re** [comment]
Hosting a full matrix instance with its DB feels a bit too much work… We'll pass on that for the moment.

**Comment by edkalrio** [comment]
I've found some cool channels in irc.2f30.org/+6697 and irc.cyberia.is/+6697

**Comment by Spydar007** [comment]
FTR, Foonetic IRC has shut down and the main XKCD channels moved to Slashnet: http://wiki.xkcd.com/irc/Main_Page#Channel_Migration

**Comment by herzenschein** [comment]
Please add irc.anthrochat.net as well. :)

**Comment by V1TA5** [comment]
Hackint please ( https://hackint.org )

**Comment by Mikaela** [comment]
Hackint is unlikely to be bridged as they explicitly forbid Matrix bridging.

> Additionally we want to express clearly, that we disallow the bridging of Matrix in general, since some of these issues are universal to its bridging stack and at the same time intolerable.

* https://hackint.org/#20181028_Matrix_Bridging_Sunset

**Comment by V1TA5** [comment]
Oh yes youre right.

**Comment by ssweeny** [comment]
I'd love to see [PSIGenix ](http://www.psigenix.net/) (irc.psigenix.net)

**Comment by almereyda** [comment]
There are also plenty of Networks mentionned on http://xdcc.it that could be compared to the ones presented here.

**Comment by BeatLink** [comment]
Rizon.net doesnt work anymore :/

**Comment by heftig** [comment]
> Rizon.net doesnt work anymore :/

Rizon turned off their bridge after the [Matrix.org break-in](https://matrix.org/blog/2019/05/08/post-mortem-and-remediations-for-apr-11-security-incident/) and didn't turn it on again.

**Comment by BeatLink** [comment]
Ahh I see. Would you guys want to edit the main description to  show that the bridge isn't up?

**Comment by SimonSapin** [comment]
> `irc.w3.org` maybe?
> 
> Or is the w3 case under the umbrella of the W3**C** comment?
> 
> > … removed W3C support due to them dropping their IPv6-enabled IRCd. …
> 
> https://www.w3.org/Project/IRC/

IPv6 appears to be fixed for `irc.w3.org`. It’d be nice for the bridge to be restored, thanks!

```
$ host -t AAAA irc.w3.org 
irc.w3.org is an alias for styx.w3.org.
styx.w3.org has IPv6 address 2603:400a:ffff:804:801e:34:0:79
$ nc -v -6 irc.w3.org 6667
Connection to irc.w3.org 6667 port [tcp/*] succeeded!

**Comment by Matombo** [comment]
What's the reason IRCnet is crossed out in the list in the first post?

**Comment by silkeh** [comment]
> What's the reason IRCnet is crossed out in the list in the first post?

Because there is an IRCnet bridge: https://github.com/matrix-org/matrix-appservice-irc/wiki/Bridged-IRC-networks

**Comment by raniesantos** [comment]
irc.spotchat.org for linux mint user please

**Comment by casperstorm** [comment]
I've been keeping an eye on this list for a few years now. 
Are there any progress on it, or can I do something myself to get a bridge to Quakenet?

**Comment by V1TA5** [comment]
Note: im not involved in running this but:

My guess is to talk to the opers on Quakenet. 
I did the same on Espernet (for my personal bridge).

Some Networks have heared of matrixes past security breach and are sceptical (or have even baned matrix from their networks like hackint has). So you should be prepared for questions about that.

You could also set up your own bridge if quakenet agrees and offer it to us here.

**Comment by weebl2000** [comment]
I know at least one oper on Qnet which is also running matrix. Worth a shot.

**Comment by pit711** [comment]
i can wait another 10 years, no problem

euirc and freechatnetwork would be nice

**Comment by jamver** [comment]
It would be nice to see an official IRC network bridge added for OzOrg, the Australian Internet Relay Chat Network.  OzOrg is a (relatively) small network, but it has been around for a very, very long time.

Web Links:
- [Oz.Org](https://www.oz.org/) (Old / Original Domain)
- [OzOrg.net](https://www.ozorg.net/) (New Domain)

Main Server:
- `irc://irc.ozorg.net:[6667-6669,7000]/` for non-SSL traffic
- `ircs://irc.ozorg.net:6697/` (SSL/TLS on port 6697)
- `irc.oz.org is an alias for irc.ozorg.net`
-- `irc.ozorg.net` has address `45.76.125.3`
-- `irc.ozorg.net` has IPv6 address `[2001:19f0:5801:19ce:5400:1ff:fecb:4ef9]`

**Comment by mhambone** [comment]
Any hope of getting Rizon to turn their brdige back on? If someone knows of a room to drop the suggestion in, I'd be happy to.

**Comment by jhulkko** [comment]
Could you please add ZJE IRC to the wishlist: irc.z.je

**Comment by baruchel** [comment]
Could you add both irc.sorcery.net and irc.darkmyst.org which are freeform roleplaying servers? Websites are https://sorcery.net/ and https://www.darkmyst.org/

**Comment by christiansacks** [comment]
+1 for EFNet

**Comment by grahamperrin** [comment]
Please use the up-vote feature of GitHub, not comments, to up-vote existing comments. Thank you.

**Comment by christiansacks** [comment]
> Please use the up-vote feature of GitHub, not additional comments, to up-vote existing comments. Thank you.
> 
> [#208 (comment)](https://github.com/matrix-org/matrix-appservice-irc/issues/208#issuecomment-259469625) is for EFnet.

I'm really sorry, I didn't know that it was a thing. How do I do that?

**Comment by grahamperrin** [comment]
> … Do you mean click the thumbs up icon …

Yes! Thank you. Thumbs-up and other icons are available through the smiley face (reaction) menu.

**Comment by jlj77** [comment]
Don't believe I've seen a suggestion for IRCNow.org. They're very interested in bridging, and I've already begun testing this with my personal homeserver. Thanks!

**Comment by ptman** [comment]
https://p.haavard.me/407 - Do we need a bridge for libera.chat?

**Comment by mvdan** [comment]
> https://p.haavard.me/407 - Do we need a bridge for libera.chat?

This is now more serious: https://www.kline.sh/

Two of the Freenode channels I frequent via the bridge have already moved to libera.chat, meaning I temporarily can't continue using them - unless I go back to a regular old IRC client.

**Comment by Kreyren** [comment]
Add libera.chat! ^_^

**Comment by ekmett** [comment]
Maybe @glguy can help out getting things started from the libera.chat side of things. Selfishly, I'd really like to see a matrix <-> libera.chat bridge for the #haskell community.

**Comment by Kreyren** [comment]
I changed my mind, I am against adding libera.chat as it's managed by the former freenode staffers that are toxic beyond believe.

**Comment by madduck** [comment]
> ~Add libera.chat! ^_^~
> 
> EDIT: i no longer believe in adding libera.chat

fortunately, your belief is pretty insignificant in the grand scheme of things 😂

**Comment by tgr** [comment]
Many of the ex-Freenode FLOSS communities are moving or have already moved to Libera.Chat ([some listed here](https://gist.github.com/shadowcat-mst/998cea12794768bdb3da2daeff31baad)). It would be great if it could be bridged soonish.

**Comment by turt2live** [comment]
A libera bridge is in progress: have patience, and join #irc:matrix.org on Matrix for updates. 

Let's try to keep this issue open for other suggestions please :)

**Comment by Mikaela** [comment]
[ErgoChat](https://ergo.chat/), `ircs://irc.ergo.chat:6697/`. It's the home of Ergo IRCd (previously known as Oragono) which is leading with implementing the IRCv3 features.

I am requesting it as I am very curious on how will matrix-appservice-irc behave when encountered some of the features such as `always-on` integrated bouncer, multiple client connections using the same nickname simultaneously, missed lines being replayed for the client including non-privmsg/non-notices by HistServ.

**Comment by utkarsh2102** [comment]
Hey,

> A libera bridge is in progress: have patience [...]

Sorry to ask again but is there any update on Matrix bridge for Libera? Or an ETA or something? TIA! 😄

**Comment by zenhack** [comment]
@utkarsh2102, see https://github.com/matrix-org/matrix-appservice-irc/issues/1324

**Comment by utkarsh2102** [comment]
> @utkarsh2102, see #1324

Awesome, thank you, Ian!

**Comment by H7-25** [comment]
It's possible request a bridge to SimosNap.com ?

**Comment by Xiretza** [comment]
Foonetic (xkcd channels) has been down and removed for a while, and #xkcd and friends have moved to [SlashNET](http://slashnet.org/). Would be nice to get a bridge for that! It supports IPV6.

**Comment by quantumgc** [comment]
Any plans to turn the Rizon bridge back on?

**Comment by xanoni** [comment]
Any updates on QuakeNet?

**Comment by ptman** [comment]
QuakeNet doesn't like bridges IIUC. Take a look at https://github.com/hifi/heisenbridge/ for bridging quakenet.

**Comment by bw4518** [comment]
Regarding Rizon, this is from their forums: 
> we've previously had a matrix bridge from matrix.org but they had to disable bridging after a security incident on their part. Bridging with matrix.org hasn't been restored since then. If matrix home server operators wish to set up a bridge with the network they can create a ticket at https://abuse.rizon.net and we'll see if we can work something out.

**Comment by Experimenter** [comment]
Requesting a bridge to Spotchat: irc.spotchat.org

**Comment by austinhuang0131** [comment]
Hackint bridge restored: https://hackint.org/#20211207_Matrix_support_is_returning_to_hackint

**Comment by varikvalefor** [comment]
An [AnthroChat](https://anthrochat.net) bridge would be cool.

**Comment by grahamperrin** [comment]
> EFnet : irc.efnet.org :zap: (high interest)

From thread <https://matrix.to/#/!OGEhHVWSdvArJzumhm:matrix.org/$lq9JyWqP6hgJx0_yTEF7PuJ3U3TaDwjinTJinjkBnsA?via=matrix.org&via=libera.chat&via=wolfo.tech>: 

> … please: realistically, when might we gain a connection to EFNet? …

– beneficiaries will include the FreeBSD Project. Via <https://www.freebsd.org/community/irc/>: <https://wiki.freebsd.org/IRC/Channels#EFnet>. 

Many thanks

**Comment by arrowd** [comment]
As a FreeBSD dev I strongly +1 for EFNet bridge. We have a lot of populated IRC channels there, which aren't reachable from Matrix for now.

**Comment by bingoxo** [comment]
can you bring back rizon please ?

**Comment by guysoft** [comment]
irc.akkoma.dev at
https://docs.akkoma.dev/develop/

**Comment by granquet** [comment]
LinuxNet at zeniv.linux.org.uk

This is where #kernel and other linux kernel related rooms are nowadays (it seems?)

**Comment by Gartral** [comment]
+1 for Libreachat please!

**Comment by t3chguy** [comment]
@Gartral libera.chat is already bridged. https://kparal.wordpress.com/2021/06/01/connecting-to-libera-chat-through-matrix/

**Comment by AssassinK786** [comment]
+1 for EFnet : irc.efnet.org please!

**Comment by F30** [comment]
Request for euIRC.net. IPv6 support is available and I can provide personal contact to an admin upon request.

**Comment by marcfargas** [comment]
Any change #perl:irc.perl.org will get there? just upvoted the comment from 2016!

**Comment by HeIIow2** [comment]
+1 for the [akkoma irc](https://docs.akkoma.dev/stable/) #akkoma-dev:irc.akkoma.dev

**Comment by SysopNiteStorm** [comment]
So I represent BSDForAll/IRCNow, would just like to add our network as an option, we support IPv6 on all servers and do support SASL authentication as well. We are a network of learning, teaching, as well as just plain old chatting. Our main channel is #IRCNow we also have several other sub-main channels, my nick on IRC is Sysop_NiteStorm.

**Comment by dmd** [comment]
slashnet?

**Comment by SysopNiteStorm** [comment]
> slashnet?

No not slashnet

**Comment by grahamperrin** [comment]
> IRCNow

<https://ircnow.org/>

**Comment by SysopNiteStorm** [comment]
correct, https://ircnow.org is the main network name, BSDForAll/AFWI we are a separate "team" within IRCNow, but also represent IRCNow if that makes sense to you.

**Comment by Sweeistaken** [comment]
I will suggest adding the new SweeNet (irc.ircat.xyz) to the list, however, I am currently going to make a Matrix bridge plugin for Codename IRCat and SweeNet.

More info: https://discuss.swee.codes/t/make-a-matrix-bridge-plugin/62

**Comment by Depresst0** [comment]
> I will suggest adding the new SweeNet (irc.ircat.xyz) to the list, however, I am currently going to make a Matrix bridge plugin for Codename IRCat and SweeNet.
> 
> More info: https://discuss.swee.codes/t/make-a-matrix-bridge-plugin/62

I would like this too.

**Comment by anarcat** [comment]
> irc.indymedia.net

for the record, I (as an IRC operator of indymedia.org (and not .net, btw)) do not wish to see our network bridged with Matrix in any shape or form. please remove or cross it off the list.

it's hard enough to deal with abuse and basic maintenance as it is, i don't think Matrix would bring us anything significant. if anything, we'd be more likely to switch the network entirely to matrix and ditch IRC than anything, but there are major blockers in having this happen:

1. E2EE actually working reliably
2. E2EE room membership (instead of relying on the server)
3. IRC-like privilege management (like "i want to kick out a dead operator", which you can't do in matrix right now)
4. proper data retention policies (like "this channel shouldn't be logged by servers at all")
5. easier high availability (like "i can spin up a VM and connect it to the network and I have a +1 in my N+1")
6. and more! see <https://anarc.at/blog/2022-06-17-matrix-notes/> and <https://telegra.ph/why-not-matrix-08-07>

some of those are design flaws in Matrix that might never be solved (3 & 4, for example), some others are implementation issues that Matrix have been trying to solve for a long time and I'm starting to be doubtful it will ever be (1 & 2)... some issues (5/6) are kind of fixed maybe if your client or server uses the right SDK/MSC and if you hold it just so. there's not "can i use" for matrix, so it's essentially impossible to see if you're secure/performant/whatever... 

so for now we're looking at keeping the network alive as an IRC network.

i also find it somewhat amusing to find this issue almost a decade later, when matrix.org is actually *retiring* all those bridges officially. i am not sure what this issue even *means* at this point...

**Comment by Xiretza** [comment]
Don't worry, this list hasn't been looked at by anyone at NV in years.

**Comment by Megver83** [comment]
BNC4Free or other public bouncers would be cool
